### ISSUE TITLE
tvOS - Support Deep Links

### ISSUE THREAD

**Issue by ebower42** [jellyfin/Swiftfin#1189]
**Describe the feature you'd like**

It would be nice to have deep link support in the tvOS app so users can utilize automations (i.e. in HomeAssistant) to open the app directly to specific media.

**Additional context** 

Something like `jellyfin://Users/{user-id}/media/{media-id}` to use as a media url for home assistant to open on the apple tv. I found the deeplink related files in the iOS app files (`Swiftfin/Objects/DeepLink.swift`). However, the main functionality for getting the media data is commented out. Is this a possible feature for the tvOS app?

**Comment by potelux** [comment]
I see this is marked as blocked, but nothing is linked as the "blocked by".  What is blocking this issue from development?

**Comment by JPKribs** [comment]
> What is blocking this issue from development?

Looks like the flag just never changed. It's been updated.

**Comment by potelux** [comment]
As far as I can tell, deep links by URL should already work.  The only issue I see is that the deeplink regex doesn't allow hyphens.

Is this issue more related to the NSUserActivity work that would allow Home Assistant to call the precise media and also pave the way for Siri and Spotlight to link to specific content?

I am interested in implementing this feature and want to be sure I am seeing the issue correctly.

EDIT: Here is the PR for the hyphen support: https://github.com/jellyfin/Swiftfin/pull/2168

**Comment by potelux** [comment]
I tested this out last night and found that the deeplinking already works.  There are few quirks and bugs, but support exists.

Here are some of the bugs:
- The deep linking does not use the navigation stack, so repeated deep link requests always return the first deep link (EDIT: PR https://github.com/jellyfin/Swiftfin/pull/2173 )
- Deep linking does not work if the app has not yet been launched
- No support exists for Apple TV Spotlight, TopShelf, etc (NSUserActivity).  So, Siri still can't launch deep links and neither can the home page of the Apple TV
